### PR TITLE
BACKENDS: timidity: Get rid of gethostbyname, use getaddrinfo.

### PR DESCRIPTION
**Pull request by susnux**
As the title says this will port the timidity code to use `getaddrinfo` instead of `gethostbyname`.
Not even that `gethostbyname` is deprecated, but also it will limit the address resolution to one possible ip (but maybe that one is not working?) and will limit the name resolution to IPv4.
That is fixed by this commit, moreover on at least openSUSE the buildsystem warns if code is using `gethostbyname`.
Also some other minor cleanup (fixed a FIXME etc).

### CODE REVIEW

File: backends/midi/timidity.cpp [hunk 1]
```diff
@@ -215,9 +205,15 @@ int MidiDriver_TIMIDITY::open() {
 	/*
 	 * open data connection
 	 */
-	data_port = atoi(res + 4);
+	num = atoi(res+4);
```
Comment (bluegr) [inline backends/midi/timidity.cpp]: Formatting (spaces around "+")

**Comment by bluegr** [comment]
Looks good overall, thanks! :)

**Comment by susnux** [comment]
Ok fixed that

File: backends/midi/timidity.cpp [hunk 2]
```diff
@@ -277,49 +273,36 @@ void MidiDriver_TIMIDITY::teardown() {
...
+	for (rp = result; rp != NULL; rp = rp->ai_next) {
+		if ((fd = socket(rp->ai_family, rp->ai_socktype, rp->ai_protocol)) == -1)
+			continue;
+		if (connect(fd, rp->ai_addr, rp->ai_addrlen) != -1)
+			break;
 		::close(fd);
+	}
+
+	if (rp == NULL) {
+		warning("TiMidity: Could not connect\n");
```
Comment (mgerhardy) [inline backends/midi/timidity.cpp]: Isn't there a small memleak here - freeaddrinfo(result) call is missing.

**Comment by susnux** [comment]
@mgerhardy Yes in some situation that could maybe leak memory.
Fixed it by moving the free above the check.

**Comment by fuzzie** [comment]
backends/midi/timidity.cpp:225:90: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘char*’ [-Wformat=]

Another format string needs updating?

**Comment by susnux** [comment]
@fuzzie  You were right, fixed it.

**Comment by sev-** [comment]
Thanks, merging.

**Pull request merged**